src/stores/businessStore.ts

```diff
@@ -91,9 +91,19 @@ export const useBusinessStore = defineStore('business', {
       return state.businessInfo.lastDirectorChangeDate
     },
 
-    /** The legal name. */
+    /** The legal name or alternate name if is firm. */
     getLegalName (state: BusinessStateIF): string {
-      return state.businessInfo.legalName
+      if (!GetFeatureFlag('enable-legal-name-fix')) {
```

Comment:
The idea here is the following:
- If FF is disabled: keep whatever we had before. Nothing changes.
- If FF is enabled:
  - We check if firm and Alternate names array. If true, this'll return name. 
  - Otherwise, just return the legal name. BENs for example.